scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/MockBigQuery.scala

```diff
@@ -60,6 +66,11 @@ class MockBigQuery private (private val bq: BigQuery) {
 
     val t = bq.tables.table(original)
     val temp = bq.tables.createTemporary(t.getLocation).getTableReference
+
+    // override project and dataset in the mutable temp object
+    stagingProject.foreach(temp.setProjectId(_))
+    stagingDataset.foreach(temp.setDatasetId(_))
```

Comment:
Wasn't this working ?
```suggestion
    stagingProject.foreach(temp.setProjectId)
    stagingDataset.foreach(temp.setDatasetId)
```